packages/hdwallet-xdefi/src/xdefi.ts

```diff
@@ -0,0 +1,288 @@
+import * as core from "@shapeshiftoss/hdwallet-core";
+import * as eth from "./ethereum";
+import _ from "lodash";
+
+class XDeFiTransport extends core.Transport {
+  public async getDeviceID() {
+    return "xdefi:0";
+  }
+
+  public async call(...args: any[]): Promise<any> {}
+}
+
+export function isXDeFi(wallet: core.HDWallet): wallet is XDeFiHDWallet {
+  return _.isObject(wallet) && (wallet as any)._isXDeFi;
+}
+
+type HasNonTrivialConstructor<T> = T extends { new (): any } ? never : T;
+
+export class XDeFiHDWallet implements core.HDWallet, core.ETHWallet {
+  readonly _supportsETH = true;
+  readonly _supportsETHInfo = true;
+  readonly _isXDeFi = true;
+
+  transport: core.Transport = new XDeFiTransport(new core.Keyring());
+  info: XDeFiHDWalletInfo & core.HDWalletInfo;
+  ethAddress?: string | null;
+  provider: any;
+
+  constructor() {
+    this.info = new XDeFiHDWalletInfo();
+  }
+
+  async getFeatures(): Promise<Record<string, any>> {
+    return {};
+  }
+
+  public async isLocked(): Promise<boolean> {
+    return !this.provider.xdefi.isUnlocked();
+  }
+
+  public getVendor(): string {
+    return "XDeFi";
+  }
+
+  public async getModel(): Promise<string> {
+    return "XDeFi";
+  }
+
+  public async getLabel(): Promise<string> {
+    return "XDeFi";
+  }
+
+  public async initialize(): Promise<any> {
+    try {
+      this.provider = (window as any).xfi && (window as any).xfi.ethereum;
+    } catch (e) {
+      console.error(e);
+    }
+  }
```

Comment:
The first overload provides a static hint that the method will throw without options given. The implementation signature complies with the core interface but isn't visible itself from other code, avoiding any ambiguity about `undefined` being a supported value.

```suggestion
  public initialize(): never;
  public initialize(provider: unknown): Promise<any>;
  public async intialize(provider?: unknown): Promise<any> {
    if (!provider) throw new Error("provider is required");
    this.provider = provider;
  }
```